srt.py

```diff
@@ -213,21 +221,10 @@ def srt_timestamp_to_timedelta(ts):
     :rtype: datetime.timedelta
     """
 
-    # This function is *extremely* hot during parsing, so please keep perf in
-    # mind.
-
-    if len(ts) < TS_LEN:
-        raise ValueError(
-            "Expected timestamp length >= {}, but got {} (value: {})".format(
-                TS_LEN, len(ts), ts
-            )
-        )
-
-    # Doing this instead of splitting based on the delimiter using re.split
-    # with a compiled regex or str.split is ~15% performance improvement during
-    # parsing. We need to look from the end because the number of hours may be
-    # >2 digits.
-    hrs, mins, secs, msecs = (int(x) for x in [ts[:-10], ts[-9:-7], ts[-6:-4], ts[-3:]])
+    match = TS_REGEX.match(ts)
+    if match is None:
+        raise TimestampParseError("Got unparseable timestamp: {}".format(ts))
```

Comment:
Hmm, this is still an SRT format, so probably can just be SRTParseError (or at least subclass it), but I can sort that out separately some time

Reply:
Ah yes, that probably would have made more sense.